app/controllers/application_letters_controller.rb

```diff
@@ -36,8 +36,8 @@ def create
 
   # PATCH/PUT /applications/1
   def update
-    if @application_letter.update(application_params)
-      redirect_to @application_letter, notice: 'Application was successfully updated.'
+    if @application_letter.update_attributes(application_params)
+      redirect_to :back, notice: 'Application was successfully updated.'
```

Comment:
How about http://stackoverflow.com/questions/771656/correctly-doing-redirect-to-back-in-ruby-on-rails-when-referrer-is-not-availabl